llvm/include/llvm/ADT/StringExtras.h

```diff
@@ -537,6 +537,9 @@ class ListSeparator {
     }
     return Separator;
   }
+
+  // Enables the use of same ListSeparator object for multiple lists.
+  void reset() { First = true; }
```

Comment:
Why not just use a new one for each list

Reply:
I did do that first, but all this printing is in a single scope, so I needed new names like DefLS, UseLS, MemOpLS etc whichi I though was just extra noise.  Reusing the same allows maintaining a canonical pattern OS << LS. But I am not too tied up and can go with that.

Reply:
Would `LS = ListSeparateor()` work to reset it?

Reply:
Done.